### ISSUE TITLE
The "coil length" value for Grover's square loop inductance equation

### ISSUE THREAD

**Issue by bkoehm** [miguelvaca/vk3cpu#1]
Thanks for this tool, such an awesome piece of work.

When calculating square loop inductance using Grover's equation, I've seen at least one reference (loudet.org, see below) that considers the "coil length" to be the width of windings on one side rather than the total length of conductor used.  This gives a significantly different result for the inductance.  I have no idea if Loudet has it right or not, but I suspect there's a chance that the "width of windings" interpretation is correct, mainly because it produces a value that seems closer to that of what I get with a LCR meter on an actual AM loop antenna I have here that has multiple turns.  I tried to find Grover's full original material online to try and disambiguate this, but so far haven't succeeded. 

Loudet web page:
https://sidstation.loudet.org/antenna-theory-en.xhtml
Equation 16 on that page is the Grover equation for a square.
An image on that web page that shows 'l' to be the width of windings rather than total length of conductor:
https://sidstation.loudet.org/05-vlf/3-antenna-theory/img/square-loop.png

**Comment by miguelvaca** [comment]
Yes, Laudet is correct. Coil length is different from conductor length. Nagaoka and Grover use the same dimensions also.
Which tool in particular were you looking at? Magloop calculator?

**Comment by bkoehm** [comment]
Yes.  I see in getInductance():
```const coil_length = cond_diameter_meters * spacing_ratio * loop_turns;
...
if(shape == "square") {
    ...
    const l = (N>1) ? (coil_length * 100.0) : (cond_diameter_meters * 100.0); // coil length in cm
    ...
}```
Where that `const l` seems like the total conductor length (when N>1) rather than the width of the coils.

Then further down I see this which I'm not sure I understand what bOn2r is, but it seems like this is based on Grover?
(The commented out //retval is certainly Grover, which is using 'l' as total coil length.  I am not sure about bOn2r.)
```
const bOn2r = l / (loop_diameter_meters * 141.4214);
                    //retval = 1e-6 * 0.008 * (N**2) * s * ( Math.log((1.4142*s*N)/((N+1)*l)) + 0.37942 + ((0.3333*(N+1)*l)/(s*N)));
                    retval = 1e-6 * 0.008 * (N**2) * s * ( Math.log(1.0/bOn2r) + 0.37942 + 0.47140*bOn2r - 0.014298*bOn2r**2 - 0.02904*bOn2r**4);
```

**Comment by miguelvaca** [comment]
Hi Brian,

"coil_length" contains the side length of the coil. Based on the conductor diameter, the spacing ratio between conductors, and finally a multiplier of conductors. (Its not the total conductor length.)

bOn2r should be read as "b-On-2r", or (b/(2*r)). Sorry if it's confusing. It's a constant that was used in a few spots in Grover's equation.
![Grover1](https://user-images.githubusercontent.com/31931809/148629105-8ab25d0e-9e08-4738-a483-478fabf1dee2.png)
![Grover2](https://user-images.githubusercontent.com/31931809/148629142-8178a32a-0ad2-4c1c-9553-b041497aded2.png)

See the attached screenshots from Grover's book. Cheers!

**Comment by bkoehm** [comment]
Ok, thank you.  You mention that coil_length contains the side length of the coil.  Is that the same thing as the "loop diameter" slider (for a square)?  I'm not clear on the difference between coil_length and the slider value for the side of the square (other than coil_length being in cm instead of m).  It seems like those two values are the same unless I am misunderstanding.  ...or, I should phrase it as: The c/a ratio I pick should result in coil_length == value of the loop diameter slider.

**Comment by miguelvaca** [comment]
Not the same thing. If looking at a coil from the side, the coil length increases as you increase the number of turns. (When I mentioned side length, I meant the length looking from the side-view of the coil. I didn't mean a straight-length of tubing that can be used with polygonal shaped antennas, which is I suppose how Grover uses it, and denotes it as "s".) The loop diameter is looking from the front, and does not change with number of turns. For a square, the loop diameter is equivalent to the width of the coil. The c/a radio (spacing ratio) is the ratio of the distance between conductor centers divided by the conductor diameter. Changing the spacing ratio changes the length of the coil, but not its diameter (or width). 

A c/a ratio of 4 or more is pretty good.

**Comment by bkoehm** [comment]
Ok, yep, back to my understanding that coil_length is the the "width" of the coil.  

I'm getting a result that makes sense now, thank you.  I suspect my original mistake was an error with the c/a ratio.  Once I confirmed what coil_length was supposed to be and what the bOn2r equation was (thanks to your explanations), then I was able to get my user input errors cleared up.  I also had to adjust some minimums and maximums in the code (I have 9 turns and I'm working with smaller gauge wire, for example).  It's giving me expected results now.

Thanks again!  Wonderful piece of work.
